From 38b52af877311ed1b42fcca89f22ed8c77568329 Mon Sep 17 00:00:00 2001
From: Vince JV <1276544+vincejv@users.noreply.github.com>
Date: Sun, 21 Apr 2024 02:43:49 +0800
Subject: [PATCH] Add versioning through activity window

---
 make/configure.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/make/configure.py b/make/configure.py
index b5a6468b9..06a8d12a6 100644
--- a/make/configure.py
+++ b/make/configure.py
@@ -1008,6 +1008,7 @@ class Project( Action ):
 
         self.url_appcast = 'https://handbrake.fr/appcast%s%s.xml' % (url_ctype,url_arch)
         self.url_appnote = 'https://handbrake.fr/appcast/%s.html' % (url_ntype)
+        self.version += '-%s-reloaded' % ('Git-Commit-Hash')
 
         self.msg_end = '%s (%s)' % (self.name,repo.type)
         self.fail = False
-- 
2.34.1

