From 4c88ffd88f8942b45562f12dae4e712094835827 Mon Sep 17 00:00:00 2001
From: Vince JV <1276544+vincejv@users.noreply.github.com>
Date: Thu, 9 May 2024 19:38:59 +0800
Subject: [PATCH] h265: Dovi compile error

---
 libhb/handbrake/h265_common.h | 28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git a/libhb/handbrake/h265_common.h b/libhb/handbrake/h265_common.h
index f434caaee..e52377f42 100644
--- a/libhb/handbrake/h265_common.h
+++ b/libhb/handbrake/h265_common.h
@@ -43,34 +43,6 @@ static const int          hb_h265_level_values[]            = {
     -1,  30,  60,  63,  90,  93, 120, 123,
     150, 153, 156, 180, 183, 186,   0, };
 
-static struct
-{
-    const char *level;
-    const int level_id;
-    const uint32_t max_luma_sample_rate;
-    const uint32_t max_luma_picture_size;
-    const uint32_t max_bitrate_main_tier;
-    const uint32_t max_bitrate_high_tier;
-}
-
-hb_h265_level_limits[] =
-{
-    { "1.0", 10, 552960,     36864,    128,    128    },
-    { "2.0", 20, 3686400,    122880,   1500,   1500   },
-    { "2.1", 31, 7372800,    245760,   3000,   3000   },
-    { "3.0", 30, 16588800,   552960,   6000,   6000   },
-    { "3.1", 31, 33177600,   983040,   10000,  10000  },
-    { "4.0", 40, 66846720,   2228224,  12000,  30000  },
-    { "4.1", 41, 133693440,  2228224,  20000,  50000  },
-    { "5.0", 50, 267386880,  8912896,  25000,  100000 },
-    { "5.1", 51, 534773760,  8912896,  40000,  160000 },
-    { "5.2", 52, 1069547520, 8912896,  60000,  240000 },
-    { "6.0", 60, 1069547520, 35651584, 60000,  240000 },
-    { "6.1", 61, 2139095040, 35651584, 120000, 480000 },
-    { "6.2", 62, 4278190080, 35651584, 240000, 800000 },
-    { NULL,  0,  0,          0,        0,      0      }
-};
-
 // stolen from libx265's x265.h
 static const char * const hb_h265_fullrange_names[] = {
     "limited", "full", NULL, };
-- 
2.34.1

