From 501f929e2b2be6d20088cc28f4d32226f7fc6922 Mon Sep 17 00:00:00 2001
From: Vince JV <1276544+vincejv@users.noreply.github.com>
Date: Mon, 6 May 2024 18:48:41 +0800
Subject: [PATCH] Bump version to `2311fba`

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 29e6c85..ca1acba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,7 @@ endif()
 project(svt-av1 VERSION 2.0.0
     LANGUAGES C CXX)
 
-set(VERSION_TAG "-c31e160")
+set(VERSION_TAG "-2311fba")
 
 if(POLICY CMP0063)
   cmake_policy(SET CMP0063 NEW)
-- 
2.34.1

