From 50310d8b4fe36c7988f1b8e8a1d1c08271906b24 Mon Sep 17 00:00:00 2001
From: Vince JV <1276544+vincejv@users.noreply.github.com>
Date: Fri, 19 Apr 2024 13:43:08 +0800
Subject: [PATCH] Allow the use of extended CRF

---
 libhb/encsvtav1.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/libhb/encsvtav1.c b/libhb/encsvtav1.c
index 381e9118a..70d904a17 100644
--- a/libhb/encsvtav1.c
+++ b/libhb/encsvtav1.c
@@ -88,6 +88,12 @@ static int alloc_buffer(EbSvtAv1EncConfiguration *config, hb_work_private_t *pv)
     return 0;
 }
 
+void set_extended_crf(EbSvtAv1EncConfiguration *param, double vquality) {
+    double decimalPart                = vquality - (int)vquality; // Get the decimal part
+    param->qp                         = vquality; // truncated automatically
+    param->extended_crf_qindex_offset = decimalPart / 25;
+}
+
 int encsvtInit(hb_work_object_t *w, hb_job_t *job)
 {
     hb_work_private_t *pv = calloc(1, sizeof(hb_work_private_t));
@@ -123,7 +129,7 @@ int encsvtInit(hb_work_object_t *w, hb_job_t *job)
     }
     else
     {
-        param->qp                = job->vquality;
+        set_extended_crf(param, job->vquality);
         param->rate_control_mode = SVT_AV1_RC_MODE_CQP_OR_CRF;
         param->force_key_frames = 1;
     }
-- 
2.34.1

