From f0d08ae105028400dfad4a543384eb2627f5c7f5 Mon Sep 17 00:00:00 2001
From: Vince JV <1276544+vincejv@users.noreply.github.com>
Date: Sat, 20 Apr 2024 23:32:16 +0800
Subject: [PATCH] Add versioning through activity window

---
 make/configure.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/make/configure.py b/make/configure.py
index b5a6468b9..cbff2d269 100644
--- a/make/configure.py
+++ b/make/configure.py
@@ -1003,7 +1003,7 @@ class Project( Action ):
                 url_ctype = '_unstable'
                 url_ntype = 'unstable'
 
-            self.build = time.strftime('%Y%m%d', now) + '00'
+            self.build = time.strftime('%Y%m%d', now) + '00-' + 'Git-Commit-Hash'
             self.title = '%s %s (%s)' % (self.name,self.version,self.build)
 
         self.url_appcast = 'https://handbrake.fr/appcast%s%s.xml' % (url_ctype,url_arch)
-- 
2.34.1

