diff --git a/contrib/svt-av1/module.defs b/contrib/svt-av1/module.defs
index 2a16c6825..9e856c742 100644
--- a/contrib/svt-av1/module.defs
+++ b/contrib/svt-av1/module.defs
@@ -13,8 +13,9 @@ SVT-AV1.CONFIGURE.args.prefix = -DCMAKE_INSTALL_PREFIX="$(SVT-AV1.CONFIGURE.pref
 SVT-AV1.CONFIGURE.deps        =
 SVT-AV1.CONFIGURE.static      =
 SVT-AV1.CONFIGURE.shared      = -DBUILD_SHARED_LIBS=OFF
-SVT-AV1.CONFIGURE.extra       = -DBUILD_DEC=OFF -DBUILD_APPS=OFF -DREPRODUCIBLE_BUILDS=1
+SVT-AV1.CONFIGURE.extra       = -DBUILD_DEC=OFF -DBUILD_APPS=OFF -DREPRODUCIBLE_BUILDS=0
 SVT-AV1.CONFIGURE.extra      += -DCMAKE_INSTALL_LIBDIR=lib
+SVT-AV1.CONFIGURE.extra      += -DENABLE_AVX512=ON -DSVT_AV1_LTO=ON
 
 ifeq ($(GCC.O),$(filter $(GCC.O),size size-aggressive))
     SVT-AV1.CONFIGURE.extra += -DCMAKE_BUILD_TYPE=MinSizeRel
